src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -852,184 +1050,6 @@ public ItemStack getTotalAmount(@Nullable final ItemStack stack)
-            {
-                final TileEntityFlowerPot tileentityflowerpot = (TileEntityFlowerPot) world.getTileEntity(pos);
-                tileentityflowerpot.readFromNBT(((JobBuilder) job).getStructure().getBlockInfo().tileentityData);
-                world.setTileEntity(pos, tileentityflowerpot);
-            }
-        }
-        else
-        {
-            if (!world.setBlockState(pos, blockState, 0x03))
-            {
-                return false;
-            }
-        }
-
-        if (block instanceof BlockChest && job instanceof JobBuilder)
-        {
-            final BlockPos buildingLocation = ((JobBuilder) job).getWorkOrder().getBuildingLocation();
-            final AbstractBuilding building = this.getOwnBuilding().getColony().getBuilding(buildingLocation);
-            building.addContainerPosition(pos);
-        }
-
-        //It will crash at blocks like water which is actually free, we don't have to decrease the stacks we have.
-        if (isBlockFree(block, block.getMetaFromState(blockState)))
-        {
-            return true;
-        }
-
-        @Nullable final ItemStack stack = BlockUtils.getItemStackFromBlockState(blockState);
-        if (stack == null || stack == ItemStack.EMPTY)
-        {
-            Log.getLogger().error("Block causes NPE: " + blockState.getBlock());
-            return false;
-        }
-
-        final List<ItemStack> itemList = new ArrayList<>();
-        itemList.add(stack);
-        if (job instanceof JobBuilder && ((JobBuilder) job).getStructure() != null
-              && ((JobBuilder) job).getStructure().getBlockInfo() != null && ((JobBuilder) job).getStructure().getBlockInfo().tileentityData != null)
-        {
-            itemList.addAll(getItemStacksOfTileEntity(((JobBuilder) job).getStructure().getBlockInfo().tileentityData));
-        }
-
-        for (final ItemStack tempStack : itemList)
-        {
-            if (tempStack != null && tempStack != ItemStack.EMPTY)
-            {
-                final int slot = worker.findFirstSlotInInventoryWith(tempStack.getItem(), tempStack.getItemDamage());
-                if (slot != -1)
-                {
-                    if (tempStack.getItem() == Items.LAVA_BUCKET)
-                    {
-                        getInventory().removeStackFromSlot(slot);
-                        getInventory().setInventorySlotContents(slot, new ItemStack(Items.BUCKET));
-                    }
-                    else
-                    {
-                        getInventory().decrStackSize(slot, 1);
-                    }
-                    reduceNeededResources(tempStack);
-                }
-            }
-        }
-
-        if (Configurations.builderBuildBlockDelay > 0 && block != Blocks.AIR)
-        {
-            setDelay(Configurations.builderBuildBlockDelay);
-        }
-
-        return true;
-    }
-
     private Boolean spawnEntity(@NotNull final Structure.StructureBlock currentBlock)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "spawnEntity" is 17 which is greater than 15 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 44 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)